pilot/pkg/model/push_context.go

```diff
@@ -763,6 +779,53 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthNPolicies(env *Environment) error {
+	authNPolicies, err := env.List(schemas.AuthenticationPolicy.Type, NamespaceAll)
+	if err != nil {
+		return err
+	}
+
+	sortConfigByCreationTime(authNPolicies)
+	ps.AuthNPolicies = map[host.Name]map[int][]*Config{}
```

Comment:
If you refactor this to a function that return (map[host.Name]map[int][]*Config, *Policy), may be we can reuse it for the other place